queries/flights.py

```diff
@@ -0,0 +1,502 @@
+
+
+_FLIGHTS_QTYPE = "Flights"
+
+
+TOPIC_LEMMAS = [
+    "flugvél",
+    "flugvöllur",
+    "flug",
+    "lenda",
+    "brottfarartími",
+    "lendingartími",
+]
+
+
+def help_text(lemma: str) -> str:
+    """Help text to return when query.py is unable to parse a query but
+    one of the above lemmas is found in it"""
+    return "Ég get svarað ef þú spyrð til dæmis: {0}?".format(
+        random.choice(
+            (
+                "Hvenær lendir næsta vél frá Kaupmannahöfn",
+                "Hvenær fer næsta flug til Lundúna",
+            )
+        )
+    )
+
+
+# This module wants to handle parse trees for queries
+HANDLE_TREE = True
+
+# The grammar nonterminals this module wants to handle
+QUERY_NONTERMINALS = {"QFlights"}
+
+# The context-free grammar for the queries recognized by this plug-in module
+GRAMMAR = """
+
+Query →
+    QFlights
+
+QFlights → QFlightsQuery '?'?
+
+QFlightsQuery →
+    QFlightsArrivalQuery
+    | QFlightsDepartureQuery
+
+QFlightsArrivalQuery →
+    QFlightsWhenArr QFlightsNextPlane QFlightsPathDesc
+
+QFlightsWhenArr →
+    QFlightsWhen "lendir"
+    | QFlightsWhen "kemur"
+    | QFlightsWhen "mætir"
+    | "hver" "er" "lendingartími"
+    | "hver" "er" "lendingartíminn" "fyrir"
+
+QFlightsDepartureQuery →
+    QFlightsWhenDepNextPlane QFlightsPathDesc
+
+QFlightsWhenDepNextPlane →
+    QFlightsWhen "leggur" QFlightsNextPlane "af" "stað"
+    > QFlightsWhenDep QFlightsNextPlane "af" "stað"
+    > QFlightsWhenDep QFlightsNextPlane
+
+QFlightsNextPlane →
+    "næsta" QFlightsPlane
+    | "næstu" QFlightsPlane
+
+QFlightsWhenDep →
+    QFlightsWhen "fer"
+    | QFlightsWhen "flýgur"
+    | "hver" "er" "brottfarartími"
+    | "hver" "er" "brottfarartíminn" "fyrir"
+
+$score(+35) QFlightsWhenDep
+
+QFlightsWhen →
+    "hvenær" | "klukkan" "hvað"
+
+$score(-35) QFlightsWhen
+
+QFlightsPlane →
+    "flug" | "flugvél" | "flugvélar" | "vél" | "vélar" | "flugs"
+
+$score(+100) QFlightsPlane
+
+QFlightsPathDesc →
+    # Specify at least one endpoint
+    QFlightsPrepLoc QFlightsPrepLoc > QFlightsPrepLoc
+
+QFlightsPrepLoc →
+    "til" QFlightsArrLoc_ef
+    | "frá" QFlightsDepLoc_þgf
+    | "í" QFlightsArrLoc_þgf
+    | "á" QFlightsArrLoc_þgf
+    | "á" QFlightsArrLoc_þf
+
+QFlightsArrLoc/fall →
+    Nl/fall
+
+QFlightsDepLoc/fall →
+    Nl/fall
+
+$tag(keep) QFlightsArrLoc/fall
+$tag(keep) QFlightsDepLoc/fall
+$score(+40) QFlights
+
+"""
+
+_LOCATION_ABBREV_MAP = {
+    "köben": "kaupmannahöfn",
+    "kef": "keflavík",
+    "álaborg": "aalborg",
+}
+
+_IATA_TO_AIRPORT_MAP = {
+    "aey": "akureyri",
+    "biu": "bíldudalur",
+    "egs": "egilsstaðir",
+    "gjr": "gjögur",
+    "gry": "grímsey",
+    "hfn": "hornafjörður",
+    "hzk": "húsavík",
+    "ifj": "ísafjörður",
+    "kef": "keflavík",
+    "rkv": "reykjavík",
+    "sak": "sauðárkrókur",
+    "tho": "þórshöfn",
+    "vey": "vestmannaeyjar",
+    "vpn": "vopnafjörður",
+}
+
+_AIRPORT_TO_IATA_MAP = {
```

Comment:
Better: `{ val: key for key, val in _IATA_TO_AIRPORT_MAP.items() }`

Reply:
Fixed!